### PR TITLE
Avoid executing OpenAPI build filters twice on build

### LINKED ISSUE #37901: OpenAPI filters run twice when storing
Yeah, so that is not ideal, I think we can get a better way to do this so that your filter is not run twice. I can look at this. Basically we need to create a document (for inclusion in the runtime), and then use that and only add runtime filters when we store.

I am about to take a break (summer holiday) so I'll only look at this next year. If you are in a hurry, you are welcome to do a PR ?

W.r.t priority, yeah that is something we can add. Having multiple filters is Quarkus specific (the spec only allow one) so from a spec p.o.w a user needs to do everything in one filter. But because we allow many we can look at a way to order them for sure

_Originally posted by @phillip-kruger in https://github.com/quarkusio/quarkus/issues/36152#issuecomment-1866009465_

### PR DESCRIPTION
**Pull request by mcruzdev**
Fixes #37901

### CODE REVIEW

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java [hunk 1]
```diff
@@ -1174,6 +1174,7 @@ private OpenApiDocument storeDocument(OutputTargetBuildItem out,
         } catch (RuntimeException re) {
             if (includeRuntimeFilters) {
                 // This is a Runtime filter, so it might not work at build time. In that case we ignore the filter.
+                log.info("running includeRuntimeFilters=true after catch RuntimeException");
```
Comment (mcruzdev) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:1177]:
```suggestion
```

**Comment by mcruzdev** [comment]
Hi @phillip-kruger, can you take a look in my solution? I am implementing tests.

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java [hunk 2]
```diff
@@ -1194,7 +1194,8 @@ private OpenApiDocument storeDocument(OutputTargetBuildItem out,
     private OpenApiDocument prepareOpenApiDocument(OpenAPI staticModel,
             OpenAPI annotationModel,
             List<AddToOpenAPIDefinitionBuildItem> openAPIBuildItems,
-            IndexView index) {
+            IndexView index,
+            Boolean executeBuildFilters) {
```
Comment (gastaldi) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:1198]:
```suggestion
            boolean executeBuildFilters) {
```

**Comment by gastaldi** [comment]
Can you please squash all the commits before we merge this?

**Comment by gastaldi** [comment]
I'd like @phillip-kruger's review before proceeding 👍🏻

**Comment by gastaldi** [comment]
BTW I renamed the PR (s/Evict/Avoid) which is what I guess you mean here. Can you also change the commit message?

**Comment by mcruzdev** [comment]
> BTW I renamed the PR (s/Evict/Avoid) which is what I guess you mean here. Can you also change the commit message?

Thank you! I will change here

**Comment by phillip-kruger** [comment]
@mcruzdev Please can you resolve conflicts, I'll have a look once that is done. Thanks :)

**Review by phillip-kruger — APPROVED** [review]

**Comment by mcruzdev** [comment]
I will fix this one tomorrow!

**Comment by mcruzdev** [comment]
Hi @phillip-kruger, I fixed the issue, I think that will work well.

**Pull request merged**